Client/Frontend/Library/HistoryPanel/HistoryPanelViewModel.swift

```diff
@@ -0,0 +1,221 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Shared
+import Storage
+import SwiftUI
+
+private class FetchInProgressError: MaybeErrorType {
+    internal var description: String {
+        return "Fetch is already in-progress"
+    }
+}
+
+class HistoryPanelViewModel: Loggable, FeatureFlagsProtocol {
+    
+    // MARK: - Properties
+    
+    private let profile: Profile
+    private let queryFetchLimit = 100
+    let historyActionables = HistoryActionablesModel.activeActionables
+    
+    private var currentFetchOffset = 0
+    var visibleSections: [Sections] = []
+    var searchTermGroups: [ASGroup<Site>] = []
+    var isFetchInProgress = false
+    var groupedSites = DateGroupedTableData<Site>()
+    
+    private var hasRecentlyClosed: Bool {
+        return profile.recentlyClosedTabs.tabs.count > 0
+    }
+    
+    let historyPanelNotifications = [ Notification.Name.FirefoxAccountChanged,
```

Comment:
Why the spacing between the brackets?

Reply:
Swiftlint would fix this automatically... 🤪

Reply:
You guys don't like it? 🥺

Reply:
It's easier on the eyes to read. But well two against one so gonna fix 😆